### PR TITLE
kcm master list api integration

### PR DESCRIPTION
**Pull request by sohailamjad12**

### CODE REVIEW

**Review by bharathravi-in — APPROVED** [review]

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html [hunk 1]
```diff
@@ -79,9 +79,15 @@
                                      {{data.columnInfo.config.categoryDisplayName}} Name
                                 </label>
                                 <mat-form-field appearance="outline" class="flex-1">
-                                    <input matInput formControlName="name" id="name" #name
+                                    <mat-select placeholder="{{app_strings.name}}" formControlName="name" id="name" #name (selectionChange)="change($event,createThemeForm)">
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html:82]: please put optional check for app_strings.name

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html [hunk 2]
```diff
@@ -79,9 +79,15 @@
...
+                                        <!-- <input #search autocomplete="off" placeholder="Search" class="assigneeSearch"
+                                          (keydown.space)="$event.stopPropagation()"  formControlName="assigneeText" matInput> -->
+                                        <mat-option *ngFor="let option of masterList"    (blur)="updateDname(option.title, createThemeForm)" [value]="option">{{option.title}}</mat-option>
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html:85]: same for option.title

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html [hunk 3]
```diff
@@ -192,6 +198,13 @@
                                             {{option.name}}
                                         </mat-option>
                                         </mat-autocomplete> -->
+                                        <mat-select placeholder="{{app_strings.name}}" formControlName="name" id="name" #name (selectionChange)="change($event,createThemeFormMulti, i )">
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html:201]: same as above

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html [hunk 4]
```diff
@@ -192,6 +198,13 @@
...
+                                            <!-- <input #search autocomplete="off" placeholder="Search" class="assigneeSearch"
+                                              (keydown.space)="$event.stopPropagation()"  formControlName="assigneeText" matInput> -->
+                                            <mat-option *ngFor="let option of masterList"    (blur)="updateDname(option.title, createThemeFormMulti, i)" [value]="option">{{option.title}}</mat-option>
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.html:204]: same as above

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 5]
```diff
@@ -206,10 +210,24 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
 
   updateFormView(form, data) {
     console.log('view',data);
-    
-    form.get('name').patchValue(data.childrenData.name)
+    console.log('array',this.masterList);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:213]: please remove console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 6]
```diff
@@ -206,10 +210,24 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
     form.get('description').patchValue(data.childrenData.description)
+
+      if (data.childrenData.name && this.masterList.length) {
+        const assignName = this.masterList.find(option =>
+          data.childrenData.name === option.title
+        )
+        if (assignName) {
+          form.controls['name'].setValue(assignName)
+        }
+        console.log(assignName);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:224]: please remove console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 7]
```diff
@@ -221,30 +239,56 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
 
   updateFormEdit(form, data) {
     console.log('ddd',data);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:241]: please remove console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 8]
```diff
@@ -221,30 +239,56 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
     form.get('dname').patchValue(data.childrenData.displayName)
     form.get('description').patchValue(data.childrenData.description)
+    if (data.childrenData.name && this.masterList.length) {
+      const assignName = this.masterList.find(option =>
+        data.childrenData.name === option.title
+      )
+      if (assignName) {
+        form.controls['name'].setValue(assignName)
+      }
+      console.log(assignName);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:251]: please remove console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 9]
```diff
@@ -221,30 +239,56 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
+      filterCriteriaMap: {
+        status: "Live",
+        isActive: true
+    },
+    requestedFields: [],
+    pageNumber: 0,
+    pagesize: 200
+    }
+    this.frameWorkService.getKcmSearchList(requestObj,this.data.columnInfo.code).subscribe((response)=>{
+      console.log('response',response);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:272]: please remove console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 10]
```diff
@@ -221,30 +239,56 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
+          
+        },1000)
+       
+
+      }
+    })
+  }
 
   multiCreate(form, data) {
     console.log('inside multiCreate')
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:294]: please remove console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 11]
```diff
@@ -254,6 +298,8 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
     if(form.valid) {
       console.log('form.valid', form.valid)
       console.log(form.value)
+      console.log('formValueTheme',form.value.themeFields);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:301]: please remove all the console

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 12]
```diff
@@ -262,11 +308,13 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
         themeFields.forEach((val, i) =>{
           const term: NSFramework.ICreateTerm = {
             code: this.frameWorkService.getUuid(),
-            name: val.name,
+            name: val.name.title,
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:311]: check if val exists then do the changes

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 13]
```diff
@@ -384,13 +432,29 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
-    }
+    // if(this.data.mode === 'create' && !form.controls['dname'].value.trim().length){
+    //   form.get('dname').patchValue(name)
+    // }
+    // if(this.data.mode === 'multi-create' && !form.controls.themeFields.controls[i].controls['dname'].value.trim().length){
+    //   form.controls.themeFields.controls[i].controls['dname'].patchValue(name)
+    // }
   }
+  change(event,form, i?){
+   if(this.data.mode === 'create'){
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:443]: check if this.data exists then apply this condition

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 14]
```diff
@@ -384,13 +432,29 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
+    form.get('dname').patchValue(event.source.value.title)
+    this.expansionTitle = event.source.value.title
+    this.cdr.detectChanges()
+
+  }
+  if(this.data.mode === 'multi-create'){
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:449]: same as above

File: projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts [hunk 15]
```diff
@@ -384,13 +432,29 @@ export class CreateTermComponent implements OnInit, AfterViewInit {
...
+    form.controls.themeFields.controls[i].controls['dname'].patchValue(event.source.value.title)
+    console.log('Updated themeFields dname:', form.controls.themeFields.controls[i].controls['dname'].value);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/components/create-term/create-term.component.ts:451]: remove console

File: projects/taxonomy-editor/src/lib/services/framework.service.ts [hunk 16]
```diff
@@ -301,8 +301,17 @@ export class FrameworkService {
     return prevSelectedTerms
   }
 
-  getKcmSearchList(requestBody:any){
-    return this.http.post(`/${this.proxiesPath}/competencyArea/search`, requestBody).pipe(map(res => _.get(res, 'result.result')))
+  getKcmSearchList(requestBody:any,category){
+     let categoryItem
+    console.log('categoryItem',category);
```
Comment (vishnubansaltarento) [inline projects/taxonomy-editor/src/lib/services/framework.service.ts:306]: remove console

**Comment by christyfernandes** [comment]
Merging this PR for now, since other developer will resolve all this comments and other issues

**Pull request merged**